### PR TITLE
[draft] doc: http keyword updates 3025 v1

### PR DESCRIPTION
**Pull request by jmtaylor90**
Make sure these boxes are signed before submitting your Pull Request -- thank you.

- [x] I have read the contributing guide lines at
   https://docs.suricata.io/en/latest/devguide/codebase/contributing/contribution-process.html
- [x] I have signed the Open Information Security Foundation contribution agreement at
   https://suricata.io/about/contribution-agreement/ (note: this is only required once)
- [x] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
https://redmine.openinfosecfoundation.org/issues/3025

Describe changes:
- as discussed in various other doc update tickets, giving this page an update and wanted to get some feedback before going too far with changes
- removed the legacy keywords and sticky buffer table as that was/is confusing
- reformatted keyword list and made list links
- updated http.method keyword with the idea of using that same format for the remaining keywords.

### Provide values to any of the below to override the defaults.

To use a pull request use a branch name like `pr/N` where `N` is the
pull request number.

Alternatively, `SV_BRANCH` may also be a link to an
OISF/suricata-verify pull-request.

```
SV_REPO=
SV_BRANCH=
SU_REPO=
SU_BRANCH=
LIBHTP_REPO=
LIBHTP_BRANCH=
```

### CODE REVIEW

**Comment by jmtaylor90** [comment]
One other question I had @jufajardini , is there a way to use the suri-rtd-test instance for doc builds to test rendering? (like you did here, https://github.com/OISF/suricata/pull/10056)

**Comment by jufajardini** [comment]
> One other question I had @jufajardini , is there a way to use the suri-rtd-test instance for doc builds to test rendering? (like you did here, #10056)

Hey there, that's just a personal instance of read the docs that I set up for testing my branches. I believe that you should be able to set up one, too. Another option would be if I (or you) pushed to my suricata repo, then I could activate that version...

**Comment by jmtaylor90** [comment]
> > One other question I had @jufajardini , is there a way to use the suri-rtd-test instance for doc builds to test rendering? (like you did here, #10056)
> 
> Hey there, that's just a personal instance of read the docs that I set up for testing my branches. I believe that you should be able to set up one, too. Another option would be if I (or you) pushed to my suricata repo, then I could activate that version...

Ah okay, I will look into that. thanks!

**Comment by jufajardini** [comment]
> > > One other question I had @jufajardini , is there a way to use the suri-rtd-test instance for doc builds to test rendering? (like you did here, #10056)
> > 
> > 
> > Hey there, that's just a personal instance of read the docs that I set up for testing my branches. I believe that you should be able to set up one, too. Another option would be if I (or you) pushed to my suricata repo, then I could activate that version...
> 
> Ah okay, I will look into that. thanks!

Cool, let me know if you get any questions about that! :)

File: doc/userguide/rules/http-keywords.rst [hunk 1]
```diff
@@ -1,180 +1,133 @@
...
+  User-Agent: Mozilla/5.0
+  Host: suricata.io
 
-.. image:: http-keywords/method.png
+.. container:: example-rule
 
-.. image:: http-keywords/Legenda_rules.png
+  alert http $HOME_NET any -> $EXTERNAL_NET any (msg:"HTTP Request Example"; \
+  flow:established,to_server; :example-rule-options:`http.method; \
+  content:"GET";`classtype:bad-unknown; sid:2; rev:1;)
```
Comment (jufajardini) [inline doc/userguide/rules/http-keywords.rst:122]: This doesn't build, I'm guessing due to a combination of the line break character with the enclosing backtick used in the `example-rule-options` selected portion of the rule being in the next line. 
This works:
```suggestion
  alert http $HOME_NET any -> $EXTERNAL_NET any (msg:"HTTP Request Example";
  flow:established,to_server; :example-rule-options:`http.method; content:"GET";`
  classtype:bad-unknown; sid:2; rev:1;)
```
:)

File: doc/userguide/rules/http-keywords.rst [hunk 2]
```diff
@@ -1,180 +1,133 @@
...
 
-.. image:: http-keywords/method1.png
```
Comment (jufajardini) [inline doc/userguide/rules/http-keywords.rst:174]: Could you please explain the removal of some of the images, but not all of them? (IMHO a general explanation could be included in the dedicated commit for removing elements from this section).

**Review by jufajardini — CHANGES_REQUESTED** [review]
Thanks for taking the stab at re-structuring this! This one will need input from others for us to decide on these changes.

Overall, one thing I'd like to see for this one is that we remove whatever we agree that could be gone (like the tables and/or images) in a single commit, and then make changes like additions in separate commits, so it's easier to review and understand the changes :)

I've left a few inline comments, but didn't do a thorough review yet, for the aforementioned reasons.

**Comment by jufajardini** [comment]
Seeing all that's going on here, being able to check what it will look like in RtD would be great, indeed.

Btw, one can check things locally - albeit they might look a bit different - by running `make html` and then accessing the generating docs from `doc/userguide/_build/html/index.html` (sorry if that was already known, sharing just cause I saw that this current version isn't compiling).

**Comment by jmtaylor90** [comment]
> Seeing all that's going on here, being able to check what it will look like in RtD would be great, indeed.
> 
> Btw, one can check things locally - albeit they might look a bit different - by running `make html` and then accessing the generating docs from `doc/userguide/_build/html/index.html` (sorry if that was already known, sharing just cause I saw that this current version isn't compiling).

I thought I had built (make html) the latest commit but apparently lost track of where I was since that fails. I have the rtd builds working so will link that as well for this work.

File: doc/userguide/rules/http-keywords.rst (hunk 1 above)
Comment (jmtaylor90) [reply doc/userguide/rules/http-keywords.rst:122]: Will be getting back to this today and get the builds sorted out.

**Comment by jmtaylor90** [comment]
https://rtd-test-suricata.readthedocs.io/en/latest/rules/http-keywords.html is the instance of rtd that builds against this branch

File: doc/userguide/rules/http-keywords.rst (hunk 2 above)
Comment (jmtaylor90) [reply doc/userguide/rules/http-keywords.rst:174]: Ah, I will be removing probably all the images but I only have reworked the HTTP primer, http keywords list and http.method to give an idea of the changes and general formatting

**Comment by catenacyber** [comment]
Thanks @jmtaylor90 for taking a stab at this :-)

**Comment by jufajardini** [comment]
> https://rtd-test-suricata.readthedocs.io/en/latest/rules/http-keywords.html is the instance of rtd that builds against this branch

Thanks for taking the time to set this up! :)

**Comment by jmtaylor90** [comment]
continuing in https://github.com/OISF/suricata/pull/10101
